python/dgl/transforms/functional.py

```diff
@@ -3300,4 +3302,69 @@ def norm_by_dst(g, etype=None):
 
     return norm
 
+def rwpe(g, k):
+    r"""Return random walk positional encodings.
+
+    For internal use.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The homogeneous graph.
+    k : int
+        The number of random walk steps.
+
+    Returns
+    -------
+    Tensor
+        The random walk positional encodings.
+    """
+    # get 1-step random walk probabilities as A * D^-1
+    A = g.adj(scipy_fmt='csr') # adjacency matrix
+    Dinv = sparse.diags(F.asnumpy(g.in_degrees()).clip(1) ** -1.0, dtype=float) # D^-1
+    RW = A * Dinv
+
+    # Iterate for k steps
+    PE = [F.tensor(RW.diagonal()).float()]
+    RW_power = RW
+    for _ in range(k-1):
+        RW_power = RW_power * RW
+        PE.append(F.tensor(RW_power.diagonal()).float())
+    PE = F.stack(PE,dim=-1)
+    
+    return PE
+
+def laplacian_pe(g, k):
+    r"""Return laplacian positional encodings.
+
+    For internal use.
```

Comment:
This API will be visible to users if you want to add it. It's just a functional interface while RWPE is a module interface.

Reply:
done